File: packages/cmk-frontend-vue/src/lib/rest-api-client/userConfig.ts
/**
 * Copyright (C) 2024 Checkmk GmbH - License: GNU General Public License v2
 * This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
 * conditions defined in the file COPYING, which is part of this source code package.
 */
import { fetchRestAPI } from '@/lib/cmkFetch'
import { API_ROOT } from './constants'

const DISMISS_WARNING_API = `${API_ROOT}/domain-types/user_config/actions/dismiss-warning/invoke`

export async function persistWarningDismissal(warning: string) {
  await fetchRestAPI(DISMISS_WARNING_API, 'POST', { warning: warning })
}

File: cmk/gui/openapi/shared_endpoint_families/user_config.py
#!/usr/bin/env python3
# Copyright (C) 2026 Checkmk GmbH - License: GNU General Public License v2
# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
# conditions defined in the file COPYING, which is part of this source code package.
from cmk.gui.openapi.restful_objects.endpoint_family import EndpointFamily

USER_CONFIG_FAMILY = EndpointFamily(
    name="Users",
    description=(
        """

Checkmk usually manages its own users, but you can also connect to external user management
systems like LDAP. The following endpoints provide a way to manage users and trigger a
synchronization of all configured user connections.

"""
    ),
    doc_group="Setup",
)
